src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAYLIST_CAPACITY = 10;
+    private int size;
+    private Object[] elements;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_ARRAYLIST_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity(size + 1);
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkBoundExclusive(index);
+        if (index == size || size >= elements.length) {
```

Comment:
Don't complicate if-else construction. The condition 'size >= elements.length' is redundant here because ensureCapacity() will already check if the array needs to grow.